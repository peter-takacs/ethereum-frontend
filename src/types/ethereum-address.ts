export class Address {
    private readonly address: string;

    public static isValid(address: string) {
        return /^0x[a-fA-F0-9]{40}$/.test(address);
    }

    public static fromBytes(bytes: Uint8Array): Address {
        if (bytes.length !== 20) {
            throw new Error(`Invalid amount of bytes supplied to ethereum address.`)
        }
        let address = '0x';
        for (const byte of bytes) {
            address = address + byte.toString(16);
        }
        return new Address(address);
    }

    constructor (address: string) {
        if (!Address.isValid(address)) {
            throw new Error(`Invalid address "${address}" supplied to ethereum address constructor.`)
        }
        this.address = address;
    }

    public toString() {
        return this.address;
    }

    public equals(other: Address | null) {
        if (other == null) {
            return false;
        }
        return this.address === other.address;
    }
}