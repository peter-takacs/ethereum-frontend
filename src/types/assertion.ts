import { Address } from "./ethereum-address";

export interface Assertion {
    certificate: string;
    issuer: Address;
}