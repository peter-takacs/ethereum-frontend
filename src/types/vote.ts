import { Address } from "./ethereum-address";

export enum VoteStatus {
    Pending = 0,
    Accept = 1,
    Reject = 2
}

export interface CandidateVotes {
    candidate: Address;
    votes: Map<Address, VoteStatus>;
}