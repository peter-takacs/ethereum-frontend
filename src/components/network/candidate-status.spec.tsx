
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { dummyAddress } from '../../utils/test/address';
import CandidateStatus from './candidate-status';
import { VoteStatus as VoteStatusEnum } from '../../types/vote' ;
import { Address } from '../../types/ethereum-address';

test('Renders correctly with no data', () => {
    const component = renderer.create(
            <CandidateStatus
                candidate={null}
                votes={new Map<Address, VoteStatusEnum>()} 
            />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
});

test('Renders correctly with data', () => {
    const votes = new Map<Address, VoteStatusEnum>();
    votes.set(dummyAddress(1), VoteStatusEnum.Accept);
    const component = renderer.create(
        <CandidateStatus
            candidate={dummyAddress(0)}
            votes={votes}
        />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})