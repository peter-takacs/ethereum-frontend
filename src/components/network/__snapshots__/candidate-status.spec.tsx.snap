// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly with data 1`] = `
<div
  className="MuiGrid-container-1"
>
  <p
    className="MuiTypography-root-98 MuiTypography-body1-107"
  >
    this.props.candidate
  </p>
</div>
`;

exports[`Renders correctly with no data 1`] = `<div />`;
